src/main/java/io/vertx/gsoc2018/qotd/QuoteOfTheDayVerticle.java

```diff
@@ -64,4 +67,36 @@ private void getAllQuotes(RoutingContext routingContext) {
       }
     });
   }
+
+  private void postNewQuote(RoutingContext routingContext) {
+    JsonObject body = routingContext.getBodyAsJson();
+    String text = body.getString("text");
+    String author = body.getString("author");
+    if (text == null) {
+      routingContext.response().setStatusCode(404)
+        .putHeader("Content-Type", "application/json; charset=utf-8")
+        .end();
+      return;
+    }
+    if (author == null) {
+      author = DEFAULT_AUTHOR;
+    }
+
+    JsonObject newQuote = new JsonObject()
+      .put("text", text)
+      .put("author", author);
+
+    databaseService.postNewQuote(newQuote, res -> {
+      if (res.succeeded()) {
+        routingContext.response().setStatusCode(200)
+          .putHeader("Content-Type", "application/json; charset=utf-8")
+          .end(newQuote.encodePrettily());
+      } else {
+        routingContext.response().setStatusCode(400)
```

Comment:
Not the right status code for a server error

Reply:
Fixed in https://github.com/tsegismont/vertx-gsoc-2018-qotd/pull/4/commits/f3f7f82f89c26b8d9789314666166bc76a9aea0f.